render/detailed/links.go

```diff
@@ -0,0 +1,230 @@
+			// These recording rules must be defined in the prometheus config.
+			// NB: Pods need to be labeled and selected by their respective Service name, meaning:
+			// - The Service's `spec.selector` needs to select on `name`
+			// - The Service's `metadata.name` needs to be the same value as `spec.selector.name`
+			docker.MemoryUsage:   parsedTemplate(`namespace_label_name:container_memory_usage_bytes:sum{label_name="{{.Label}}"}`),
+			docker.CPUTotalUsage: parsedTemplate(`namespace_label_name:container_cpu_usage_seconds_total:sum_rate{label_name="{{.Label}}"}`),
+		},
+	}
+	k8sControllers = map[string]struct{}{
+		report.Deployment: {},
+		"stateful_set":    {},
+		"cron_job":        {},
+	}
+)
+
+// SetMetricsGraphURL sets the URL we deduce our eventual metric link from.
+// Supports placeholders such as `:orgID` and `:query`. An empty url disables
+// this feature. If the `:query` part is missing, a JSON version will be
+// appended, see `queryParamsAsJSON()` for more info.
+func SetMetricsGraphURL(url string) {
+	metricsGraphURL = url
+}
+
+// RenderMetricURLs sets respective URLs for metrics in a node summary. Missing metrics
+// where we have a query for will be appended as an empty metric (no values or samples).
+func RenderMetricURLs(summary NodeSummary, n report.Node) NodeSummary {
+	if metricsGraphURL == "" {
+		return summary
+	}
+
+	queries := getTopologyQueries(n.Topology)
+	if len(queries) == 0 {
+		return summary
+	}
+
+	var maxprio float64
+	var bs bytes.Buffer
+	var ms []report.MetricRow
+	found := make(map[string]struct{})
+
+	// Set URL on existing metrics
+	for _, metric := range summary.Metrics {
+		if metric.Priority > maxprio {
+			maxprio = metric.Priority
+		}
+		tpl := queries[metric.ID]
+		if tpl == nil {
+			continue
+		}
+
+		bs.Reset()
+		if err := tpl.Execute(&bs, summary); err != nil {
+			continue
+		}
+
+		ms = append(ms, metric)
+		ms[len(ms)-1].URL = buildURL(bs.String())
+		found[metric.ID] = struct{}{}
+	}
+
+	// Append empty metrics for unattached queries
+	for _, metadata := range shownQueries {
+		if _, ok := found[metadata.ID]; ok {
+			continue
+		}
+
+		tpl := queries[metadata.ID]
+		if tpl == nil {
+			continue
+		}
+
+		bs.Reset()
+		if err := tpl.Execute(&bs, summary); err != nil {
+			continue
+		}
+
+		maxprio++
+		ms = append(ms, report.MetricRow{
+			ID:         metadata.ID,
+			Label:      metadata.Label,
+			URL:        buildURL(bs.String()),
+			Metric:     &report.Metric{},
+			Priority:   maxprio,
+			ValueEmpty: true,
+		})
+	}
+
+	summary.Metrics = ms
+
+	return summary
+}
+
+// buildURL puts together the URL by looking at the configured `metricsGraphURL`.
+func buildURL(query string) string {
+	if strings.Contains(metricsGraphURL, urlQueryVarName) {
+		return strings.Replace(metricsGraphURL, urlQueryVarName, url.QueryEscape(query), -1)
+	}
+
+	params, err := queryParamsAsJSON(query)
+	if err != nil {
+		return ""
+	}
+
+	if metricsGraphURL[len(metricsGraphURL)-1] != '/' {
```

Comment:
Yes, we bail early in https://github.com/weaveworks/scope/pull/2664/files#diff-7a21e593fb345568b7d0bc0774375c97R112